### PR TITLE
feat(template): add workflow bot csharp template

### PR DESCRIPTION
**Pull request by dooriya**
E2E TEST: N/A

### CODE REVIEW

File: templates/bot/csharp/workflow/GettingStarted.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-------------------------
+1. Download and install ngrok (https://ngrok.com/)
+2. Open a command prompt and run ngrok http 5130
+3. Right-click your project and select Teams Toolkit > Prepare Teams app dependencies
+4. If prompted, sign in with an M365 account for the Teams organization you want 
+to install the app to
+5. Press F5, or select the Debug > Start Debugging menu in Visual Studio
+6. In the launched browser, select the Add button to load the app in Teams
+7. In the chat bar, type and send "helloWorld" to your app to trigger a response
+
```
Comment (yukun-dong) [inline templates/bot/csharp/workflow/GettingStarted.txt:13]: Better to add tutorial for "doStuff" as well.

File: templates/bot/csharp/workflow/AdapterWithErrorHandler.cs.tpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                // NOTE: In production environment, you should consider logging this to
+                // Azure Application Insights. Visit https://aka.ms/bottelemetry to see how
+                // to add telemetry capture to your bot.
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encountered an error or bug.");
+                await turnContext.SendActivityAsync("To continue to run this bot, please fix the bot source code.");
+
+                // Send a trace activity, which will be displayed in the Bot Framework Emulator
```
Comment (yukun-dong) [inline templates/bot/csharp/workflow/AdapterWithErrorHandler.cs.tpl:24]: Maybe just "send a trace activity". Bot Framework Emulator won't be used here right?

File: templates/bot/csharp/workflow/GettingStarted.txt (hunk 1 above)
Comment (dooriya) [reply templates/bot/csharp/workflow/GettingStarted.txt:13]: Let me add the aka link to the action handler tutorial here.

File: templates/bot/csharp/workflow/AdapterWithErrorHandler.cs.tpl (hunk 2 above)
Comment (dooriya) [reply templates/bot/csharp/workflow/AdapterWithErrorHandler.cs.tpl:24]: will remove the `Bot Framework Emulator` part since it's not used in a Teams bot

**Review by yukun-dong — DISMISSED** [review]

File: templates/bot/csharp/workflow/CardActions/DoStuffActionHandler.cs.tpl [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿using AdaptiveCards.Templating;
+using Microsoft.Bot.Builder;
+using Microsoft.TeamsFx.Conversation;
+using Newtonsoft.Json;
+using {{SafeProjectName}}.Models;
```
Comment (swatDong) [inline templates/bot/csharp/workflow/CardActions/DoStuffActionHandler.cs.tpl:5]: Consolidate namespace order in all cs files. (Here project namespace is at bottom but in the handler cs file it's at top...)

File: templates/bot/csharp/workflow/{{SafeProjectName}}.csproj.tpl [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <ProjectCapability Include="TeamsFx" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Include=".fx/**/*" />
+    <None Remove="build/**/*" />
+    <Content Remove="build/**/*" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (swatDong) [inline templates/bot/csharp/workflow/{{SafeProjectName}}.csproj.tpl:18]: TeamsFx SDK?

Comment (dooriya) [reply templates/bot/csharp/workflow/{{SafeProjectName}}.csproj.tpl:18]: Good catch, will fix later

**Review by swatDong — APPROVED** [review]

**Review by Siglud — APPROVED** [review]

**Pull request merged**